plugins/social_network/social_network.cpp

```diff
@@ -89,6 +171,213 @@ namespace golos { namespace plugins { namespace social_network {
+                                auto result = dmp.patch_apply(patch, utf8_to_wstring(to_string(con.body)));
+                                auto patched_body = wstring_to_utf8(result.first);
+                                if(!fc::is_utf8(patched_body)) {
+                                    idump(("invalid utf8")(patched_body));
+                                    from_string(con.body, fc::prune_invalid_utf8(patched_body));
+                                }
+                                else {
+                                    from_string(con.body, patched_body);
+                                }
+                            }
+                            else { // replace
+                                from_string(con.body, o.body);
+                            }
+                        } catch ( ... ) {
+                            from_string(con.body, o.body);
+                        }
+                    }
+                    // Set depth null if needed (this parameter is given in config)
+                    if (depth_parameters.set_null_after_update) {
+                        con.block_number = db.head_block_num();
+                    } 
+                });
+
+            }
+            else {
+                // Creation case
+
+                const auto &new_comment = db.get_comment(o.author, o.permlink);
+                comment_id_type id = new_comment.id;
+
+                db.create<comment_content_object>([&](comment_content_object& con) {
+                    con.comment = id;
+                    if (!depth_parameters.has_comment_title_depth || depth_parameters.comment_title_depth > 0) {
+                        from_string(con.title, o.title);
+                    }
+                    
+                    if ((!depth_parameters.has_comment_body_depth || depth_parameters.comment_body_depth > 0) && o.body.size() < 1024*1024*128) {
+                        from_string(con.body, o.body);
+                    }
+
+                    if ((!depth_parameters.has_comment_json_metadata_depth || depth_parameters.comment_json_metadata_depth > 0) &&
+                        fc::is_utf8(o.json_metadata)
+                    ) {
+                        from_string(con.json_metadata, o.json_metadata);
+                    }
+                    else {
+                        wlog("Comment ${a}/${p} contains invalid UTF-8 metadata",
+                            ("a", o.author)("p", o.permlink));
+                    }
+
+                    con.block_number = db.head_block_num();
+                });
+            }
+        }
+
+        // Checking should we delete comment_content if needed depth has been expired
+        void operator()(const comment_payout_update_operation& o) const { 
+            if (!depth_parameters.need_clear()) {
+                return;
+            }
+
+            const auto &comment = db.get_comment(o.author, o.permlink);
+            auto& content = get_comment_content(comment.id);
```

Comment:
The content can absent, so you should use `find_comment_content` and check result on `nullptr`